Embed RunWipeHistory into Modpack QoL Menu

Zyruvias opened the pull request:
- Add Cgull's RunWipeHistory Mod, accessible from the QoL Menu in the Modpack
- Change RunWipeHistory access from the "Assist" Button to QoL Menu to prevent text from showing up mid-run,

Museus committed:
v1.2.0: update version string and Changelog

Zyruvias committed:
add ModUtil to gitignore

Zyruvias committed:
add ModUtil to gitignore

Zyruvias committed:
add Zyruvias to contributor notes :croven:

Zyruvias committed:
add UI for RunHistoryWipe

Zyruvias committed:
add RunHistoryWipe (by cgull)

cgu11 commented:
we should probably redo this as a git submodule the same way we have many of the others, given that RunHistoryWipe has its own repo.

cgu11 commented on .gitignore:
```diff
@@ -3,3 +3,4 @@
 ################################################################################
 
 /.vs
+ModUtil/**
```
why was this needed? not an issue more just curious

Zyruvias replied:
It's not needed, just an artifact of me floundering around before I learned what submodules on GitHub actually are and how they worked.

> we should probably redo this as a git submodule the same way we have many of the others, given that RunHistoryWipe has its own repo.

Sounds good, I will work on getting that set up.

Zyruvias committed:
fix gitignore

Zyruvias committed:
add submodule

Zyruvias commented:
@cgu11 This should be ready to go now, I have a pull request up on your repo and the submodule is set up. Let me know if you have any other feedback.

Zyruvias committed:
Merge branch 'ellomenop:main' into main

Zyruvias committed:
Merge branch 'main' into zyru/embed-savefile-elixir

Zyruvias committed:
remote old runHistoryWipe

Zyruvias committed:
revert gitignore

Zyruvias committed:
setting: treat all non-completion Deaths as QuickRestarts

Setting is added under QuickRestart in mod configuration menu

Reviewed-by: Museus <69139347+Museus@users.noreply.github.com>

Museus committed:
build(deps): update ModUtil from 2021Jun30 to v2.7.0

Kenorah committed:
refactor(deprecated): update ModUtil calls to newest version

Fix: #80

SleepSoul committed:
(feat)(MinibossControl: Add Hypermodded ruleset, which removes Tisiphone)

Adds A_Boss01-03 as options in all rulesets (all always true, except A_Boss03 being disabled in Hypermodded). If a Fury is invalid for the current ruleset, it will be marked ineligible as a room after A_PreBoss01 (endshop); editing LinkedRooms directly does not work.

Fix: #86
Co-authored-by: SleepSoul <SleepSoul#6006>
Co-authored-by: Museus <museus@protonmail.com>

SleepSoul committed:
feat(ChaosControl): Add ChaosControl mod, allowing certain blessings and curses to be added or removed depending on selected ruleset.
Uses new Table.MergeKeyed func from ModUtil 2.8.0 as Table.Merge does not remove remaining curses from LootData table.
No config UI as yet, will be written when new UI lib is complete; also remove Debug ruleset at this point as it only exists for testing and should not be displayed.

Co-authored-by: SleepSoul#6006
Co-authored-by: Museus <museus@protonmail.com>

SleepSoul committed:
build(deps): update ModUtil to v2.8.0

SleepSoul committed:
(refactor)(MinibossControl): Simplify Fury code using ModUtil v2.8.0 function Table.MergeKeyed; Remove previous Path.Wrap

SleepSoul committed:
build(deps): update ModUtil to v2.8.1

SleepSoul committed:
feat(EnemyControl): Add EnemyControl mod, allowing enemy pools to be defined on a per-biome basis, or enemies to be removed from the vanilla pools.

SleepSoul committed:
feat(SatyrSackControl): Add option to force short tunnels

Wraps IsRoomEligible to mark the small rooms as ineligible if 3 rooms or higher into a tunnel.

SleepSoul committed:
refactor(SatyrSackControl): Replace Path.Override with Path.Wrap for both features
Path.Wrap is simpler and avoids any future messiness with other mods that need to edit IsRoomEligible and IsRoomForced.

Author: Zyruvias#3283

Zyruvias committed:
use wraps

SleepSoul committed:
feat(RCLib): Add RCLib mod

Standardises functions used in ChaosControl and EnemyControl, as well as a lookup table for readable alternatives to internal names.

SleepSoul committed:
refactor(EnemyControl): Update to use RCLib

Add limited RuleOverride function to force enemies as eligible,
Remove inbuilt functions/lookup table and replace with references to RCLib,
Refactor also adds ability to edit Hades adds and use SuperElite enemies

SleepSoul committed:
refactor(ChaosControl): Update to use RCLib

Remove inbuilt functions and replace with references to RCLib,
Refactor also adds vanilla table inheritance and readable names for traits

SleepSoul committed:
fix(EnemyControl): Fix EnemyControl crash when RuleOverrides are not set for current preset

Add an extra nil check, as previous one would itself cause a crash if there were no overrides at all for a preset.

Zyruvias committed:
chore: fix merge conflicts on QoL menu code

Museus commented:
Cherry-picked this into `main-v130` to save us all the headache of rebasing it
